### PR TITLE
feat: activate Focal Loss, add Dice Loss + span-width weighting, OpenVINO INT8 pipeline

### PR DESCRIPTION
**Pull request by ALI-AL-MARJANI**
### 1. Loss functions (`gliner/modeling/loss_functions.py`, `base.py`, `trainer.py`)
  
  `focal_loss_with_logits` already exists in the codebase but is disabled by default
  (`alpha=-1, gamma=0`). This PR:
  - Exposes `loss_type`, `focal_loss_alpha`, `focal_loss_gamma` in `TrainingArguments` so
  users can activate it
  - Adds `span_dice_loss()` — span-level Dice Loss adapted from Li et al. (ACL 2020),
  applied element-wise over the `(B, L×K, T)` logit tensor with `ignore_index` masking
  - Adds `use_span_width_weight` flag: positive spans of width k receive `w(k) = 1 +
  log(k+1)` — zero inference overhead
  
  **Motivation:** WNUT-17 has 187× more negative spans than positive entities (0.53%
  positive ratio). BCE's gradient is dominated by trivial negatives. Focal α=0.25 delivers
  +0.99 pp WNUT-17 F1; Dice delivers +0.70 pp.
  
  ### 2. Bug fixes (`gliner/utils.py`, `gliner/modeling/encoder.py`, `gliner/model.py`,
  `gliner/onnx/model.py`)
  
  - `is_module_available()` changed from `__import__()` to `importlib.util.find_spec()` —
  prevents optional packages (peft, tensorflow) from being eagerly imported, which caused
  OpenMP deadlocks on macOS ARM
  - `encoder.py`: `kwargs.pop("token_lengths", None)` prevents crash on bi-encoder models
  that pass this GLiNER-internal kwarg to HuggingFace forward methods
  - Lazy import of `Trainer`/`TrainingArguments` in `model.py` — avoids importing
  torch.distributed at module load time
  
  ### 3. OpenVINO INT8 pipeline (`scripts/convert_to_openvino.py`)
  
  New script: ONNX → OpenVINO IR → INT8 weight compression via
  `nncf.compress_weights(INT8_ASYM)`.
  - 2.35× CPU speedup, 4× model size reduction, no accuracy degradation
  - Uses weight-only compression (not activation quantization) because GLiNER's ONNX graph
  contains `If` nodes with dynamic rank that the CPU plugin rejects during calibration-based
   quantization 
  
  ## Benchmark
  
  | Config | WNUT-17 F1 | Δ |
  |---|---|---|
  | BCE baseline | 50.09% | — |
  | Focal α=0.25, γ=2 | **51.08%** | +0.99 pp |
  | Dice Loss | 50.79% | +0.70 pp |
  
  | Backend | Latency | Speedup | Size |
  |---|---|---|---|
  | PyTorch FP32 | 59ms | 1× | 721MB |
  | OpenVINO INT8 | **25ms** | **2.35×** | **181MB** |

  Model: `knowledgator/gliner-bi-small-v1.0`, 200 fine-tuning steps on CoNLL-2003, eval on
  WNUT-17.

### CODE REVIEW

File: gliner/onnx/model.py [hunk 1]
```diff
@@ -5,14 +5,18 @@
...
+from __future__ import annotations
 
 import warnings
 from abc import ABC, abstractmethod
 from typing import Any, Dict, Optional
 
 import numpy as np
 import torch
-import onnxruntime as ort
+try:
```
Comment (Ingvarstep) [inline gliner/onnx/model.py]: Please, use https://github.com/urchade/GLiNER/blob/dfc00617594e6d4b24a6ef793f2b0efaf507ef67/gliner/utils.py#L61 instead, so we have consistency over the repo.

**Review by Ingvarstep — COMMENTED** [review]
Thank you very much for your great contribution. I have a few suggestions:

1. First, use importing utils so the project is consistent from this perspective;
2. Move ablation scripts into scripts/dice_loss_study, or something like that;

**Comment by ALI-AL-MARJANI** [comment]
- Consistency: gliner/onnx/model.py now uses is_module_available() instead of the local try/except around the onnxruntime import. While at it, found the same onnxruntime import duplicated in gliner/model.py itself (an unguarded try/except shadowed by the existing is_module_available()-gated block further down) — removed the dead one. Also found that block had been changed to hardcode ONNX_AVAILABLE = True even without onnxruntime installed; traced the actual export_to_onnx() call chain and confirmed it only needs torch.onnx.export, never ort — so the behavior was actually correct, just left a dead/misleading precondition check behind. Removed that, restored ONNX_AVAILABLE to truthfully reflect availability.

- Scripts: moved baseline_eval.py, convert_to_openvino.py, train_ablation.py, visualize_results.py into scripts/dice_loss_study/ — they're four numbered steps of one self-contained pipeline, not independent scripts.

Also merged latest main (52 commits — the branch was quite stale) to pick up the streaming feature and clear the merge conflict. Along the way restored a docstring that had been accidentally dropped from create_training_args during the lazy-import fix for the macOS deadlock, and fixed the ruff errors introduced by that version drift (same story as the other two PRs — pip install ruff with no version pin, ended up on 0.16.0 mid-review).
Ruff clean, 340 tests passing. Let me know if this all looks right
